google-cloud-logging/src/main/java/com/google/cloud/logging/Logging.java

```diff
@@ -597,6 +597,57 @@ public static TailOption project(String project) {
    */
   boolean deleteLog(String log);
 
+  /**
+   * Deletes a log and all its log entries for given log destination (see 'logName' parameter in
+   * https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry). The log will reappear if
+   * new entries are written to it.
+   *
+   * <p>Example of deleting a log.
+   *
+   * <pre>
+   * {
+   *   &#64;code
+   *   String logName = "my_log_name";
+   *   String folder = "my_folder";
+   *   boolean deleted = logging.deleteLog(logName, LogDestinationName.folder(folder));
+   *   if (deleted) {
+   *     // the log was deleted
+   *   } else {
+   *     // the log was not found
+   *   }
+   * }
+   * </pre>
+   *
+   * @return {@code true} if the log was deleted, {@code false} if it was not found
+   */
+  boolean deleteLog(String log, LogDestinationName destination);
```

Comment:
nit: consider providing default definition. this allows to handle a case when users derive from the interface in some way but then upgrade to the next minor version of the library.